js/5etools-vehicles.js

```diff
@@ -0,0 +1,230 @@
+		DataUtil.loadJSON(fluffUrl).then((data) => {
+			d20plus.vehicles.fluff = data.vehicleFluff;
+		}).catch(e => {
+			// eslint-disable-next-line no-console
+			console.error(e);
+		});
+	}
+
+	// Given the action name and entries, add an action
+	d20plus.vehicles._addAction = function (character, renderer, actionName, entries, numActions) {
+		const name = d20plus.importer.getCleanText(renderer.render(actionName));
+		const text = d20plus.importer.getCleanText(renderer.render({entries: entries}, 1)).replace(/^\s*Hit:\s*/, "");
+		d20plus.importer.addVehicleAction(character, name, text, numActions);
+	}
+
+	// Sets the image, mostly stolen from Giddy
+	d20plus.vehicles._setFluffImage = function (character, data, fluff) {
+		const tokenUrl = `${IMG_URL}vehicles/tokens/${data.source}/${data.name}.png`;
+		const avatar = data.tokenUrl || Parser.nameToTokenName(tokenUrl);
+		console.log(avatar);
+		const firstFluffImage = d20plus.cfg.getOrDefault("import", "importCharAvatar") === "Portrait (where available)" && fluff && fluff.images ? 
+			(() => {
+			const firstImage = fluff.images[0] || {};
+			return (firstImage.href || {}).type === "internal" ? `${BASE_SITE_URL}/img/${firstImage.href.path}` : (firstImage.href || {}).url;
+		})() : null;
+		$.ajax({
+			url: avatar,
+			type: "HEAD",
+			error: function () {
+				d20plus.importer.getSetAvatarImage(character, `${IMG_URL}blank.png`, firstFluffImage);
+			},
+			success: function () {
+				d20plus.importer.getSetAvatarImage(character, avatar, firstFluffImage);
+			},
+		});
+	}
+
+	// Sets fluff text, mostly stolen from Giddy
+	d20plus.vehicles._setFluff = function (character, data, fluff, renderer) {
+		let renderFluff;
+		if (data.entries) {
+			renderFluff = renderer.render({entries: data.entries}, 1);
+		}
+		if (fluff && data.hasFluff) {
+			const depth = fluff.type === "section" ? -1 : 2;
+			if (fluff.type !== "section") renderer.setFirstSection(false);
+			renderFluff = renderer.render({type: fluff.type, entries: fluff.entries || []}, depth);
+		}
+
+		if (renderFluff) {
+			setTimeout(() => {
+				const fluffAs = d20plus.cfg.get("import", "importFluffAs") || d20plus.cfg.getDefault("import", "importFluffAs");
+				let k = fluffAs === "Bio" ? "bio" : "gmnotes";
+				character.updateBlobs({
+					[k]: Markdown.parse(renderFluff),
+				});
+				character.save({
+					[k]: (new Date()).getTime(),
+				});
+			}, 500);
+		}
+	}
+
+	// Import Vehicle button was clicked
+	d20plus.vehicles.button = function () {
+		const url = $("#import-vehicles-url").val();
+
+		if (url && url.trim()) {
+			DataUtil.loadJSON(url).then((data) => {
+				d20plus.importer.addBrewMeta(data._meta);
+				d20plus.importer.showImportList(
+					"vehicle",
+					data.vehicle,
+					d20plus.vehicles.handoutBuilder,
+				);
+			});
+		}
+	};
+
+	d20plus.vehicles.handoutBuilder = function (data, overwrite, inJournals, folderName, saveIdsTo, options) {
+		// First make sure the fluff is there
+		d20plus.vehicles._loadFluff();
```

Comment:
`_loadFluff` should really be an async method which only returns when the fluff is loaded, and `await`ed here. I'm fairly this `handoutBuilder` method can be made `async` in order to enable this, as it's just used as a callback